### PR TITLE
Issue 2490 - Can't import MPAS data lacking explicit time coordinate variable

### LINKED ISSUE #2490: Can't import MPAS data lacking explicit time coordinate variable
VAPOR can't import data that do not have an explicit time coordinate variable.

To reproduce:
1. Try to import the following files from data/Source/MPAS/RosimarRiosBerrios:

history.2000-05-31_00.00.00.nc x5.tropical_3km_10N.init.nc

Note, both files must be imported. The x5 file contains the mesh info.

### PR DESCRIPTION
**Pull request by clyne**
Fixed #2490

### CODE REVIEW

File: lib/vdc/DCMPAS.cpp [hunk 1]
```diff
@@ -1156,28 +1156,49 @@ int DCMPAS::_InitCoordvars(
...
-        timeDimName, ncdfc, xTimeVarName, timeDimName
-    );
+	DC::CoordVar cvarInfo;
+	if (ncdfc->VariableExists(ncdfc->GetNumTimeSteps()-1, xTimeVarName)) {
 
-	rc = derivedVar->Initialize();
-	if (rc<0) return(-1);
+		// Create and install the Time coordinate variable
+		//
+		DerivedCoordVar_WRFTime *derivedVar = new DerivedCoordVar_WRFTime(
```
Comment (StasJ) [inline lib/vdc/DCMPAS.cpp:1171]: Did you intend to use the derived WRF time here as opposed to the new derived time class?

File: lib/vdc/DerivedVar.cpp [hunk 2]
```diff
@@ -1615,6 +1615,113 @@ bool DerivedCoordVar_TimeInSeconds::VariableExists(
...
+//	DerivedCoordVar_Time
+//
+////////////////////////////////////////////////////////////////////////////// 
+
+DerivedCoordVar_Time::DerivedCoordVar_Time(
+	string derivedVarName, string dimName, size_t n
+) : DerivedCoordVar(derivedVarName) {
+
+	_times.clear();
+	for (size_t i=0; i<n; i++) _times.push_back((float) i);
```
Comment (StasJ) [inline lib/vdc/DerivedVar.cpp:1629]: Isn't this just an array of 0..n where a[i] = i? Its not a big deal but I just wanted to make sure this is intentional.

File: lib/vdc/DCMPAS.cpp (hunk 1 above)
Comment (clyne) [reply lib/vdc/DCMPAS.cpp:1171]: Yes, the new class is used below.

File: lib/vdc/DerivedVar.cpp (hunk 2 above)
Comment (clyne) [reply lib/vdc/DerivedVar.cpp:1629]: Correct.

**Review by StasJ — APPROVED** [review]

**Comment by sgpearse** [comment]
@clyne Are you able to verify the bug has been fixed?  I'm getting an error on Casper.

Screen Shot 2020-11-20 at 7 01 24 AM

**Comment by clyne** [comment]
> @clyne Are you able to verify the bug has been fixed? I'm getting an error on Casper.
> 
> Screen Shot 2020-11-20 at 7 01 24 AM

Yes, I am. The error message says it's out of memory. How much memory did you request for the node? FYI a single 3D variable + coordinates is about ~100GBs.

**Comment by sgpearse** [comment]
I've tested with 200GB so far, using the command below.  Requesting more pushes me back in the queue and my jobs are taking some time to be issued.  I'm currently trying to submit a 300G job to Casper, but it's taking a long time to start.

vncmgr create --account nvst0001 --time 2:00:00 --job-opts="--mem=200G"

**Pull request merged**